pyaedt/edb_grpc/core/edb_data/padstacks_data.py

```diff
@@ -0,0 +1,1656 @@
+        self.edb_padstack.SetData(newPadstackDefinitionData)
+
+    @property
+    def hole_plating_thickness(self):
+        """Hole plating thickness.
+
+        Returns
+        -------
+        float
+            Thickness of the hole plating if present.
+        """
+        if len(self.hole_properties) > 0:
+            return (float(self.hole_properties[0]) * self.hole_plating_ratio / 100) / 2
+        else:
+            return 0
+
+    @hole_plating_thickness.setter
+    def hole_plating_thickness(self, value):
+        """Hole plating thickness.
+
+        Returns
+        -------
+        float
+            Thickness of the hole plating if present.
+        """
+        hr = 200 * float(value) / float(self.hole_properties[0])
+        self.hole_plating_ratio = hr
+
+    @property
+    def hole_finished_size(self):
+        """Finished hole size.
+
+        Returns
+        -------
+        float
+            Finished size of the hole (Total Size + PlatingThickess*2).
+        """
+        if len(self.hole_properties) > 0:
+            return float(self.hole_properties[0]) - (self.hole_plating_thickness * 2)
+        else:
+            return 0
+
+    @property
+    def material(self):
+        """Hole material.
+
+        Returns
+        -------
+        str
+            Material of the hole.
+        """
+        return self.edb_padstack.GetData().GetMaterial()
+
+    @material.setter
+    def material(self, materialname):
+        originalPadstackDefinitionData = self.edb_padstack.GetData()
+        newPadstackDefinitionData = self._edb.Definition.PadstackDefData(originalPadstackDefinitionData)
+        newPadstackDefinitionData.SetMaterial(materialname)
+        self.edb_padstack.SetData(newPadstackDefinitionData)
+
+    @property
+    def padstack_instances(self):
+        """Get all the vias that belongs to active Padstack definition.
+
+        Returns
+        -------
+        dict
+        """
+        return {
+            id: via for id, via in self._ppadstack.padstack_instances.items() if via.padstack_definition == self.name
+        }
+
+    @pyaedt_function_handler()
+    def convert_to_3d_microvias(self, convert_only_signal_vias=True, hole_wall_angle=15):
+        """Convert actual padstack instance to microvias 3D Objects with a given aspect ratio.
+
+        Parameters
+        ----------
+        convert_only_signal_vias : bool, optional
+            Either to convert only vias belonging to signal nets or all vias. Defaults is ``True``.
+        hole_wall_angle : float, optional
+            Angle of laser penetration in deg. It will define the bottom hole diameter with the following formula:
+            HoleDiameter -2*tan(laser_angle* Hole depth). Hole depth is the height of the via (dielectric thickness).
+            The default value is ``15``.
+            The bottom hole will be ``0.75*HoleDepth/HoleDiam``.
+
+        Returns
+        -------
+        bool
```

Comment:
```suggestion
        """Convert actual padstack instance to microvias 3D Objects with a given aspect ratio.

        Parameters
        ----------
        convert_only_signal_vias : bool, optional
            Either to convert only vias belonging to signal nets or all vias. Defaults is ``True``.
        hole_wall_angle : float, optional
            Angle of laser penetration in deg. It will define the bottom hole diameter with the following formula:
            HoleDiameter -2*tan(laser_angle* Hole depth). Hole depth is the height of the via (dielectric thickness).
            The default value is ``15``.
            The bottom hole will be ``0.75*HoleDepth/HoleDiam``.

        Returns
        -------
        bool
```
```suggestion
        """Convert the padstack instance to microvia 3D objects with a given aspect ratio.

        Parameters
        ----------
        convert_only_signal_vias : bool, optional
            Whether to convert only vias belonging to signal nets. The default is
            ``True``. If ``False``, vias belonging to all nets are converted.
        hole_wall_angle : float, optional
            Angle of laser penetration in deg. It will define the bottom hole diameter with the following formula:
            HoleDiameter -2*tan(laser_angle* Hole depth). Hole depth is the height of the via (dielectric thickness).
            The default value is ``15``.
            The bottom hole will be ``0.75*HoleDepth/HoleDiam``.

        Returns
        -------
        bool
```